platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -606,6 +606,47 @@ bool OfflineDatabase::putTile(const Resource::TileData& tile,
     return true;
 }
 
+std::exception_ptr OfflineDatabase::invalidateTileCache() try {
+    // clang-format off
+    mapbox::sqlite::Query query{ getStatement(
+        "UPDATE tiles "
+        "SET expires = 0, must_revalidate = 1 "
+        "WHERE id NOT IN ("
+        "    SELECT tile_id FROM region_tiles"
+        ")"
+    ) };
+    // clang-format on
+
+    query.run();
+    return nullptr;
+} catch (const mapbox::sqlite::Exception& ex) {
```

Comment:
question: shall we catch all the exceptions? Is the contract for this API `noexcept`? if yes, we could add `noexcept` to the method declaration and catch everything inside, if not, maybe we should not catch `mapbox::sqlite::Exception&` either... :thinking: